test/fixtures.ts

```diff
@@ -1,122 +1,124 @@
-import { Model, Config, attr, hasMany, belongsTo, hasOne } from '../src/index';
-
-class ApplicationRecord extends Model {
-  static baseUrl = 'http://example.com';
-  static apiNamespace = '/api';
+import { 
+  JSORMBase, 
+  Model, 
+  ModelConstructor, 
+  Config, 
+  attr, 
+  hasMany, 
+  belongsTo, 
+  hasOne 
+} from '../src/index';
+
+import { 
+  Attr,
+  BelongsTo,
+  HasMany,
+  HasOne,
+} from '../src/decorators'
+
+@Model({
+  baseUrl: 'http://example.com',
+  apiNamespace: '/api'
+})
+export class ApplicationRecord extends JSORMBase {
 }
 
-// typescript class
-class Person extends ApplicationRecord {
+@Model()
+export class Person extends ApplicationRecord {
   static endpoint = '/v1/people';
   static jsonapiType = 'people';
 
-  firstName: string = attr();
-  lastName: string = attr();
+  @Attr firstName : string
+  @Attr lastName : string
 }
 
-class PersonWithExtraAttr extends Person {
-  extraThing: string = attr({ persist: false });
+@Model()
+export class PersonWithExtraAttr extends Person {
+  @Attr({persist: false}) extraThing: string;
 }
 
-class PersonWithoutCamelizedKeys extends Person {
-  static camelizeKeys = false;
-
-  first_name: string = attr();
+@Model({camelizeKeys: false})
+export class PersonWithoutCamelizedKeys extends Person {
+  @Attr first_name: string
 }
 
-// Ensure setup() can be run multiple times with no problems
-// putting this here, otherwise relations wont be available.
-Config.setup();
+@Model({
+  endpoint: '/v1/authors',
+  jsonapiType: 'authors',
+})
+export class Author extends Person {
+  @Attr nilly : string
+  @HasMany({type: 'multi_words'}) multiWords : MultiWord[]
+  @HasMany('books') specialBooks : Book[]
+  @HasMany() books : Book[]
+  @HasMany() tags : Tag[]
+  @BelongsTo({type: 'genres'}) genre : Genre
+  @HasOne('bios') bio : Bio
+}
 
-// plain js class
-let Author = Person.extend({
-  static: {
-    endpoint: '/v1/authors',
-    jsonapiType: 'authors'
-  },
+@Model()
+export class Book extends ApplicationRecord {
+  static jsonapiType = 'books';
 
-  nilly:        attr(),
+  @Attr title: string
 
-  multiWords:   hasMany('multi_words'),
-  specialBooks: hasMany('books'),
-  books:        hasMany(),
-  tags:         hasMany(),
-  genre:        belongsTo('genres'),
-  bio:          hasOne('bios')
-});
+  @BelongsTo({type: 'genres'}) genre : Genre
+  @HasOne({type: Author}) author : any
+}
 
-let NonFictionAuthor = Author.extend({
+export const NonFictionAuthor = Author.extend({
   static: {
     endpoint: '/v1/non_fiction_authors',
     jsonapiType: 'non_fiction_authors',
     camelizeKeys: false
   },
 
-  nilly:         attr(),
+  attrs: {
```

Comment:
Was there a specific reason for this?

Reply:
pretty much the only way to get reliable types output for the `extend()` API, unfortunately.

Reply:
worth it